src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3372,6 +3374,13 @@ void c_typecheck_baset::typecheck_side_effect_assignment(
 
 void c_typecheck_baset::make_constant(exprt &expr)
 {
+  // Floating-point expressions may require a rounding mode.
+  // The compile-time rounding mode may differ from the
+  // run-time rounding mode.
+  const auto rounding_mode =
+    from_integer(config.ansi_c.rounding_mode, signed_int_type());
+  adjust_float_expressions(expr, rounding_mode);
```

Comment:
```
// ISO 9899:1999
// 6.6.2 A constant expression can be evaluated during translation rather than runtime, and accordingly may be used in any place that a constant may be.
//
// F.7.2 During translation the IEC 60559 default modes are in effect:
// The rounding direction mode is round to nearest
```

I can't think of a way of changing the rounding mode during translation phase so this should be round to nearest (ties to even).

Reply:
Say IBM's XL C let's you set the compile time rounding mode on the command line:
https://www.ibm.com/support/knowledgecenter/SSGH2K_11.1.0/com.ibm.xlc111.aix.doc/compiler_ref/opt_y_lower.html
But will change to "round to nearest".